packages/pcd/pod-ticket-pcd/src/PODTicketPCDPackage.ts

```diff
@@ -0,0 +1,244 @@
+    },
+    attendeeName: {
+      type: "string",
+      value: ticket.attendeeName
+    },
+    ticketId: {
+      type: "string", // should we convert UUIDs to bigints here?
+      value: ticket.ticketId
+    },
+    eventId: {
+      type: "string",
+      value: ticket.eventId
+    },
+    productId: {
+      type: "string",
+      value: ticket.productId
+    },
+    attendeeSemaphoreId: {
+      type: "cryptographic",
+      value: BigInt(ticket.attendeeSemaphoreId)
+    },
+    eventName: {
+      type: "string",
+      value: ticket.eventName
+    },
+    ticketName: {
+      type: "string",
+      value: ticket.ticketName
+    },
+    isConsumed: {
+      type: "int",
+      value: ticket.isConsumed ? 1n : 0n
+    },
+    isRevoked: {
+      type: "int",
+      value: ticket.isRevoked ? 1n : 0n
+    },
+    checkerEmail: {
+      type: "string",
+      value: ticket.checkerEmail ?? ""
+    },
+    timestampConsumed: {
+      type: "int",
+      value: BigInt(ticket.timestampConsumed)
+    },
+    timestampSigned: {
+      type: "int",
+      value: BigInt(ticket.timestampSigned)
+    },
+    imageUrl: {
+      type: "string",
+      value: ticket.imageUrl ?? ""
+    },
+    imageAltText: {
+      type: "string",
+      value: ticket.imageAltText ?? ""
+    },
+    ticketCategory: {
+      type: "int",
+      value: BigInt(ticket.ticketCategory)
+    }
+  };
+}
+
+/**
+ * Creates a new {@link PODTicketPCD} by generating an {@link PODTicketPCDProof}
+ * and deriving an {@link PODTicketPCDClaim} from the given {@link PODTicketPCDArgs}.
+ */
+export async function prove(args: PODTicketPCDArgs): Promise<PODTicketPCD> {
+  if (!args.privateKey.value) {
+    throw new Error("missing private key");
+  }
+
+  if (!args.ticket.value) {
+    throw new Error("missing ticket value");
+  }
+
+  const ticket = args.ticket.value;
+
+  const podPCD = await PODPCDPackage.prove({
+    privateKey: args.privateKey,
+    id: {
+      value: undefined,
+      argumentType: ArgumentTypeName.String
+    },
+    entries: {
+      value: ticketDataToEntries(ticket),
+      argumentType: ArgumentTypeName.Object
+    }
+  });
+
+  const id = args.id.value ?? uuid();
+
+  return new PODTicketPCD(id, { ticket }, { podPCD });
+}
+
+/**
+ * Verifies a POD Ticket PCD by checking that its {@link PODTicketPCDClaim} corresponds to
+ * its {@link PODTicketPCDProof}. If they match, the function returns true, otherwise false.
+ * In most cases, verifying the validity of the PCD with this function is not enough.
+ * It may also be necessary to ensure that the parameters of the ticket, such as the
+ * productId and eventId, match the expected values, and that the public key of the
+ * entity that signed the ticket is indeed the authority for that event.
+ */
+export async function verify(pcd: PODTicketPCD): Promise<boolean> {
+  const content = PODContent.fromEntries(ticketDataToEntries(pcd.claim.ticket));
+
+  if (content.contentID !== pcd.proof.podPCD.pod.contentID) {
```

Comment:
The intent here is that I want to ensure that the data in the `PODTicketPCD`'s claim matches the data in the `PODPCD`'s entries. Rather than comparing value-by-value, it seems better to build equivalent `PODContent` objects and compare their content IDs. However, this may not be maximally efficient, and there may be better ways to do this. For instance, should I be unpacking the content ID of the `PODPCD` contained in the proof from the signature, rather than having the `PODPCD` regenerate it?

Although `pcd.proof.podPCD.pod.contentID` looks like a property access, it's actually a get accessor, which might be constructing the content ID by building a Merkle tree.